include/fizzy/fizzy.h

```diff
@@ -275,6 +275,28 @@ uint8_t* fizzy_get_instance_memory_data(FizzyInstance* instance);
 /// @note    Function returns memory size regardless of whether memory is exported or not.
 size_t fizzy_get_instance_memory_size(FizzyInstance* instance);
 
+/// Find exported function by name.
+///
+/// @param  instance        Pointer to instance.
+/// @param  name            The function name. NULL-terminated string. Cannot be NULL.
+/// @param  out_function    Pointer to output struct to store the found function. Cannot be NULL.
+///                         If function is found, associated context is allocated, which must exist
+///                         as long as the function can be called by some other instance, and should
+///                         be destroyed with fizzy_free_exported_function afterwards.
+///                         When function is not found (false returned), this out_function is not
+///                         modified, and fizzy_free_exported_function must not be called.
+/// @returns                true if function was found, false otherwise.
+bool fizzy_find_exported_function(
+    FizzyInstance* instance, const char* name, FizzyExternalFunction* out_function);
+
+/// Free resources associated with exported function.
+///
+/// If passed pointer is NULL, has no effect.
+///
+/// @note   This function must be called only with external functions returned from
+///         fizzy_find_exported_function.
+void fizzy_free_exported_function(const FizzyExternalFunction* external_function);
```

Comment:
I wonder if we should actually make this work on a copy (`fizzy_free_exported_function(FizzyExternalFunction)`), given the only thing we're freeing is a pointer in the struct.

From an API PoV I think the current one makes more sense, but it may not be fully intuitive for the user?

Reply:
Not sure, it's a function operating on a struct, so we pass it by const pointer to avoid unnecessaty copying, seems usual for C ?

Reply:
I mean the fact we don't obtain it as a pointer, but rather allocate it on the stack in the caller.

Reply:
Yeah I see, I'm not sure what's better.

Another option is to call it with `_context` anyway: `fizzy_free_exported_function_context(FizzyExternalFunction func)` - this might be less weird than passing an instance copy to a function that's supposed to "free function".

Reply:
According to comment https://github.com/wasmx/fizzy/pull/633#discussion_r521299948 @chfast seems to prefer `FizzyExternalFunction*` argument